Use Poetry for package management (#914)
anitab-org/mentorship-backend: README.md

 - [Fork, Clone and Remote](https://github.com/anitab-org/mentorship-backend/wiki/Fork%2C-Clone-%26-Remote)
 - [Export Environment Variables](docs/environment-variables.md)

### Run app locally

> The project runs on Python 3.9. We use [Poetry](https://python-poetry.org/) to manage dependencies. It's powerful yet simple to use.

Reviewer: @bartekpacia we should test it for other python versions as well. Python 3.9 is launched recently and not all users have migrated to it

Author: I agree, however, we shouldn't hold onto the old versions as well.
I will test on 3.8.6, 3.7.9 and 3.6.12

Reviewer: Yes, just test them from python>=3.6

Author: @devkapilbansal done :)

 - [Fork, Clone and Remote](https://github.com/anitab-org/mentorship-backend/wiki/Fork%2C-Clone-%26-Remote)
 - [Export Environment Variables](docs/environment-variables.md)

### Run app locally

The project runs on Python 3.9. We use [Poetry](https://python-poetry.org/) to manage dependencies. It's powerful yet simple to use.
Here are a few basic commands you'll use the most:
- `poetry install` - installs the dependencies from `pyproject.toml` package
- `poetry add <package>` - adds the `<package>` to `pyproject.toml` and installs it
- `poetry remove <package>`- removes the `<package>` from `pyproject.toml` and uninstalls it
- `poetry shell` - spawns a shell within the virtual environment. Equivalent of `virtualenv activate`, for those familiar with [`virtualenv`](https://pypi.org/project/virtualenv/)

> - `deactivate` - deactivates the virtual environemnt. Counterpart for `poetry shell`. Use this when you finish work.

Reviewer 2: could you put these in a subsection about managing dependencies perhaps, it looks confusing being just together with the steps. being in a separate and clear section would be easier to skim through.

Author: I redesigned this section a bit. Let me know what you think:)

Reviewer 2: Looks good :) I just added a comment about "improving" the text that makes the intro to the poetry commands.

Understand more about our technical decisions made along with this project development in [Technical Decisions Wiki page](https://github.com/anitab-org/mentorship-backend/wiki/Technical-Decisions).

### Poetry commands

> As said earlier, we use Poetry. It's simple yet powerful, but we know that many people use other package management tools.

Reviewer 2: ```suggestion
As said earlier, we use [Poetry](https://python-poetry.org/).
```
Small suggestion, I removed the 2nd part of the sentence to make this intro to the commands shorter. With regards to documentation, it's important to be concise right? (that is why the run step was reduced again to 1 mini section that supports both windows and linux/mac run) So there's no need to leave opinionated text 🤔 It seems like filler.

Author: You're right, that sentence doesn't add much value

> To get you up to speed with Poetry, below we listed a few basic commands you'll use the most:

Reviewer 2: ```suggestion
To get you up to speed, here's a few basic commands you'll use the most:
```
